# modelo logit 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.iolib.summary2 import summary_col

In [2]:
GEIH_logit = pd.read_csv("Logit.csv")
GEIH_logit

Unnamed: 0  DIRECTORIO  SECUENCIA_P  ORDEN    Sexo JefeHogar    Etnia  \
0                0     5000000            1      1  Hombre    JEFEHO  Ninguno   
1                1     5000001            1      1  Hombre    JEFEHO  Ninguno   
2                2     5000001            1      2   Mujer         2  Ninguno   
3                3     5000001            1      3  Hombre         3  Ninguno   
4                4     5000002            1      1   Mujer    JEFEHO  Ninguno   
...            ...         ...          ...    ...     ...       ...      ...   
382456       30500     7309048            1      2  Hombre         2  Ninguno   
382457       30501     7309048            1      3  Hombre         3  Ninguno   
382458       30502     7309049            1      1   Mujer    JEFEHO  Ninguno   
382459       30503     7309050            1      1  Hombre    JEFEHO  Ninguno   
382460       30504     7309050            1      2   Mujer         2  Ninguno   

       NivelEducativo  FactorExpansion Departamento  ...  \
0        Bachillerato      1432.463323    Atlantico  ...   
1            Primaria      1088.796266    Atlantico  ...   
2            Primaria      1088.796266    Atlantico  ...   
3            Primaria      1088.796266    Atlantico  ...   
4        Bachillerato      2066.712422    Atlantico  ...   
...               ...              ...          ...  ...   
382456   Bachillerato       365.671281    Risaralda  ...   
382457  Universitaria       365.671281    Risaralda  ...   
382458    Tecnologica        25.636473      Vichada  ...   
382459       Primaria        25.167948      Vichada  ...   
382460   Bachillerato        25.167948      Vichada  ...   

        Departamento_Putumayo  Departamento_Quindio  Departamento_Risaralda  \
0                           0                     0                       0   
1                           0                     0                       0   
2                           0                     0                       0   
3                           0                     0                       0   
4                           0                     0                       0   
...                       ...                   ...                     ...   
382456                      0                     0                       1   
382457                      0                     0                       1   
382458                      0                     0                       0   
382459                      0                     0                       0   
382460                      0                     0                       0   

        Departamento_SAI  Departamento_Santander  Departamento_Sucre  \
0                      0                       0                   0   
1                      0                       0                   0   
2                      0                       0                   0   
3                      0                       0                   0   
4                      0                       0                   0   
...                  ...                     ...                 ...   
382456                 0                       0                   0   
382457                 0                       0                   0   
382458                 0                       0                   0   
382459                 0                       0                   0   
382460                 0                       0                   0   

        Departamento_Tolima  Departamento_Valle  Departamento_Vaupes  \
0                         0                   0                    0   
1                         0                   0                    0   
2                         0                   0                    0   
3                         0                   0                    0   
4                         0                   0                    0   
...                     ...                 ...                  ...

Escenario base: Hombre, no es jefe de hogar, no se identifica con alguna Etnia, vive en Bogotá, y tiene educación universitaria.

In [3]:
x = GEIH_logit[['Sexo_Mujer',
       'JefeHogar_JEFEHO', 'Etnia_Gitano', 'Etnia_Indígena',
       'Etnia_Negro_mulato_afrod_afroc', 'Etnia_Palenquero',
       'Etnia_Raizal_SAI',
       'Departamento_Amazonas', 'Departamento_Antioquia',
       'Departamento_Arauca', 'Departamento_Atlantico',
       'Departamento_Bolivar', 'Departamento_Boyaca', 'Departamento_Caldas',
       'Departamento_Caqueta', 'Departamento_Casanare', 'Departamento_Cauca',
       'Departamento_Cesar', 'Departamento_Choco', 'Departamento_Cordoba',
       'Departamento_Cundinamarca', 'Departamento_Guainia',
       'Departamento_Guajira', 'Departamento_Guaviare', 'Departamento_Huila',
       'Departamento_Magdalena', 'Departamento_Meta', 'Departamento_Narino',
       'Departamento_NorteSantander', 'Departamento_Putumayo',
       'Departamento_Quindio', 'Departamento_Risaralda', 'Departamento_SAI',
       'Departamento_Santander', 'Departamento_Sucre', 'Departamento_Tolima',
       'Departamento_Valle', 'Departamento_Vaupes', 'Departamento_Vichada','NivelEducativo_B_tecnico', 'NivelEducativo_Bachillerato',
       'NivelEducativo_Doctorado', 'NivelEducativo_Especializacion',
       'NivelEducativo_Maestria', 'NivelEducativo_Ninguno',
       'NivelEducativo_Normalista', 'NivelEducativo_Preescolar',
       'NivelEducativo_Primaria', 'NivelEducativo_Secundaria',
       'NivelEducativo_Tecnica', 'NivelEducativo_Tecnologica']]
y = GEIH_logit["Ocupacion_informal"]

logit = sm.Logit(y,x).fit()
print(logit.summary())

Optimization terminated successfully.
         Current function value: 0.606982
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:     Ocupacion_informal   No. Observations:               382461
Model:                          Logit   Df Residuals:                   382410
Method:                           MLE   Df Model:                           50
Date:                Fri, 01 Dec 2023   Pseudo R-squ.:                  0.1193
Time:                        08:15:20   Log-Likelihood:            -2.3215e+05
converged:                       True   LL-Null:                   -2.6361e+05
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Sexo_Mujer                        -0.2326      0.007    -32.856     

In [4]:
print(logit.get_margeff().summary())


        Logit Marginal Effects       
Dep. Variable:     Ocupacion_informal
Method:                          dydx
At:                           overall
                                    dy/dx    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Sexo_Mujer                        -0.0487      0.001    -33.024      0.000      -0.052      -0.046
JefeHogar_JEFEHO                  -0.0137      0.001     -9.239      0.000      -0.017      -0.011
Etnia_Gitano                       0.0757      0.061      1.235      0.217      -0.044       0.196
Etnia_Indígena                     0.0143      0.005      3.140      0.002       0.005       0.023
Etnia_Negro_mulato_afrod_afroc    -0.0141      0.003     -4.209      0.000      -0.021      -0.008
Etnia_Palenquero                  -0.0034      0.042     -0.082      0.935      -0.085       0.079
Etnia_Raizal_SAI                  -0.0037      0.015    

Variables NO estadísticamente significativas : 
Etnia_Gitano  Pvalor=0.217  ;     
Etnia_Palenquero Pvalor= 0.935  ;
Etnia_Raizal_SAI  Pvalor=  0.801; 
Departamento_Guajira Pvalor=  0.062 ;    
Departamento_Magdalena  Pvalor=   0.852 ;
Departamento_Narino   Pvalor=  0.205  ;
Departamento_Sucre     Pvalor=  0.155 ;
NivelEducativo_Tecnologica Pvalor= 0.141;
NivelEducativo_Normalista   Pvalor= 0.293 







In [5]:
summary= {"Sexo_Mujer":-0.0487,     
"JefeHogar_JEFEHO":-0.0137,     
"Etnia_Gitano":0.0757,     
"Etnia_Indígena":0.0143,     
"Etnia_Negro_mulato_afrod_afroc":-0.0141,   
"Etnia_Palenquero":-0.0034,     
"Etnia_Raizal_SAI":-0.0037,      
"Departamento_Amazonas":-0.0950,     
"Departamento_Antioquia":-0.1811,     
"Departamento_Arauca":-0.0786,   
"Departamento_Atlantico":-0.0840,     
"Departamento_Bolivar":-0.0243, 
"Departamento_Boyaca":-0.0841,     
"Departamento_Caldas":-0.2089 ,    
"Departamento_Caqueta":-0.0259 ,    
"Departamento_Casanare":-0.0558 ,     
"Departamento_Cauca":-0.0608 ,     
"Departamento_Cesar":-0.0272  ,   
"Departamento_Choco":-0.0648  ,    
"Departamento_Cordoba":-0.0623 ,     
"Departamento_Cundinamarca":-0.1842 ,     
"Departamento_Guainia":-0.0238  ,   
"Departamento_Guajira":-0.0086  ,    
"Departamento_Guaviare":-0.0574 ,     
"Departamento_Huila":-0.0659 ,      
"Departamento_Magdalena":-0.0008  ,    
"Departamento_Meta":-0.0860   ,   
"Departamento_Narino":-0.0050  ,    
"Departamento_NorteSantander":0.0239 ,     
"Departamento_Putumayo":-0.1504  ,    
"Departamento_Quindio":-0.1638   ,  
"Departamento_Risaralda":-0.1831  ,    
"Departamento_SAI":-0.2081  ,    
"Departamento_Santander":-0.0854  ,    
"Departamento_Sucre":0.0062,     
"Departamento_Tolima":-0.0859,     
"Departamento_Valle":-0.1453,     
"Departamento_Vaupes":-0.2291 ,     
"Departamento_Vichada":-0.0772  ,    
"NivelEducativo_B_tecnico":0.1359  ,   
"NivelEducativo_Bachillerato":0.1852   ,  
"NivelEducativo_Doctorado":-0.4713  ,    
"NivelEducativo_Especializacion": -0.2332 ,     
"NivelEducativo_Maestria": -0.3366  ,    
"NivelEducativo_Ninguno": 0.3877  ,   
"NivelEducativo_Normalista" : -0.0185 ,    
"NivelEducativo_Preescolar" : 0.4521  ,   
"NivelEducativo_Primaria" :0.3486  ,    
"NivelEducativo_Secundaria" : 0.3240 ,     
"NivelEducativo_Tecnica"  : 0.0215 ,      
"NivelEducativo_Tecnologica": -0.0059 }    

In [6]:
Summary_logit= pd.DataFrame(list(summary.items()), columns=["Dummy", "dy/dx"])
Summary_logit.to_csv("summarylogit.csv")
Summary_logit

Dummy   dy/dx
0                       Sexo_Mujer -0.0487
1                 JefeHogar_JEFEHO -0.0137
2                     Etnia_Gitano  0.0757
3                   Etnia_Indígena  0.0143
4   Etnia_Negro_mulato_afrod_afroc -0.0141
5                 Etnia_Palenquero -0.0034
6                 Etnia_Raizal_SAI -0.0037
7            Departamento_Amazonas -0.0950
8           Departamento_Antioquia -0.1811
9              Departamento_Arauca -0.0786
10          Departamento_Atlantico -0.0840
11            Departamento_Bolivar -0.0243
12             Departamento_Boyaca -0.0841
13             Departamento_Caldas -0.2089
14            Departamento_Caqueta -0.0259
15           Departamento_Casanare -0.0558
16              Departamento_Cauca -0.0608
17              Departamento_Cesar -0.0272
18              Departamento_Choco -0.0648
19            Departamento_Cordoba -0.0623
20       Departamento_Cundinamarca -0.1842
21            Departamento_Guainia -0.0238
22            Departamento_Guajira -0.0086
23           Departamento_Guaviare -0.0574
24              Departamento_Huila -0.0659
25          Departamento_Magdalena -0.0008
26               Departamento_Meta -0.0860
27             Departamento_Narino -0.0050
28     Departamento_NorteSantander  0.0239
29           Departamento_Putumayo -0.1504
30            Departamento_Quindio -0.1638
31          Departamento_Risaralda -0.1831
32                Departamento_SAI -0.2081
33          Departamento_Santander -0.0854
34              Departamento_Sucre  0.0062
35             Departamento_Tolima -0.0859
36              Departamento_Valle -0.1453
37             Departamento_Vaupes -0.2291
38            Departamento_Vichada -0.0772
39        NivelEducativo_B_tecnico  0.1359
40     NivelEducativo_Bachillerato  0.1852
41        NivelEducativo_Doctorado -0.4713
42  NivelEducativo_Especializacion -0.2332
43         NivelEducativo_Maestria -0.3366
44          NivelEducativo_Ninguno  0.3877
45       NivelEducativo_Normalista -0.0185
46       NivelEducativo_Preescolar  0.4521
47         NivelEducativo_Primaria  0.3486
48       NivelEducativo_Secundaria  0.3240
49          NivelEducativo_Tecnica  0.0215
50      NivelEducativo_Tecnologica -0.0059

-Ser mujer disminuye la probabilidad de ser informal en un 4%  
-Ser jefe de hogar disminuye la probabilidad de ser informal en un 1%  
-Ser indígena aumenta  la probabilidad de ser informal en 1%  
-Ser afro o mulato aumenta la probabilidad de ser informal en un 1%  
-Con respecto a los departamentos, aquellos que disminuyen la probabilidad de ser informal en mayor medida son Antioquia, Caldas, Valle y Vaupes, y por el contrario, ser de Norte de Santander aumenta la probabilidad de serlo en un 2%.  
-Tener doctorado, maestría y especialización disminuyen la probabilidad de ser informal en más de un 20%. En contrapartida, haber alcanzado solo hasta bachillerato aumenta la probabilidad en más de un 10% 